### PR TITLE
Vue: Add Way To Build Package

### PR DESCRIPTION
**Pull request by ErikCH**
*Issue #, if available:* [Asana](https://app.asana.com/0/0/1200663219572565/f)

*Description of changes:*
Created build for Vue and setup tests with Jest

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/core/src/types/authMachine.ts [hunk 1]
```diff
@@ -33,6 +33,7 @@ export interface SignUpContext {
   login_mechanisms?: string[];
   intent?: string;
   authAttributes?: Record<string, any>;
+  challengeName?: string;
 }
```
Comment (ErikCH) [inline packages/core/src/types/authMachine.ts:37]: When I compiled my Vue, it had errors that the types weren't correct for some of these contexts. I had to add these so it would compile. Since they are all optional it should be OK.

File: package.json [hunk 2]
```diff
@@ -32,7 +32,9 @@
     "@mdx-js/mdx/**/trim": "^0.0.3",
     "fs-extra": "^10.0.0",
     "next": "^11.0.1",
-    "ws": "^7.4.6"
+    "ws": "^7.4.6",
+    "jest": "^26.6.3",
+    "ts-jest": "^26.5.6"
```
Comment (ErikCH) [inline package.json:37]: Vue test utils doesn't work with Jest 27, so I had to lock this at 26. The React tests seems to still work with it, so this should be fine.

File: packages/vue/package.json [hunk 3]
```diff
@@ -1,13 +1,46 @@
...
+    "dist"
+  ],
   "scripts": {
-    "build": "echo \"Skipped\"",
-    "test": "echo \"Skipped: no test specified\""
+    "prebuild": "rimraf dist",
+    "build": "vite build",
+    "dev": "vite build --watch",
+    "clean": "rimraf dist",
+    "test": "jest __tests__"
```
Comment (reesscot) [inline packages/vue/package.json:20]: Maybe add a test:watch command as well?

Comment (ErikCH) [reply packages/vue/package.json]: OK! I'll add the same ones as React

File: packages/vue/src/components/sign-up.vue [hunk 4]
```diff
@@ -16,9 +16,6 @@
           <user-name-alias />
           <sign-up-password-control />
           <sign-up-confirm-password-control />
-          <base-box data-ui-error v-if="error">
-            {{ error }}
-          </base-box>
```
Comment (wlee221) [inline packages/vue/src/components/sign-up.vue:21]: Why was this removed?

File: packages/core/src/types/authMachine.ts (hunk 1 above)
Comment (wlee221) [reply packages/core/src/types/authMachine.ts:37]: This is probably on me! But I prefer we don't add `challengeName` to other types, as this context is only necessary in signIn. Where have you found these errors?

**Review by wlee221 — COMMENTED** [review]
Thanks @ErikCH! Just got few nits, but changes look good to me otherwise.

File: packages/vue/src/components/sign-up.vue (hunk 4 above)
Comment (ErikCH) [reply packages/vue/src/components/sign-up.vue:21]: This was derived from sold old code that was on line 150 for a watcher. With the new redesign this actually was broke, I believe. There is already an error tracked elsewhere on this page, so I don't think we need 2 of them.

**Review by wlee221 — APPROVED** [review]

File: packages/vue/src/components/sign-up.vue (hunk 4 above)
Comment (ErikCH) [reply packages/vue/src/components/sign-up.vue:21]: For now, let's leave the `challengeName` in the TypeScript file. For some reason the plugin I'm using likes it. Let's set a follow up task to troubleshoot it for the future.

File: packages/core/src/types/authMachine.ts (hunk 1 above)
Comment (wlee221) [reply packages/core/src/types/authMachine.ts:37]: Resolving as discussed, we'll tackle this later.
